applications/HDF5Application/custom_io/hdf5_file.cpp

```diff
@@ -372,253 +427,158 @@ void File::AddPath(const std::string& rPath)
-    {
-        DeleteAttribute(rObjectPath, rName);
+    if constexpr(local_dimension == 0) {
+        space_id = H5Screate(H5S_SCALAR);
+    } else {
+        std::vector<hsize_t> shape(local_dimension);
+        type_traits::Shape(rValue, shape.data(), shape.data() + local_dimension);
+        space_id = H5Screate_simple(local_dimension, shape.data(), nullptr);
     }
-    type_id = H5T_NATIVE_CHAR;
-    const std::array<hsize_t, 1> dims = {rValue.size()};
-    space_id = H5Screate_simple(dims.size(), dims.data(), nullptr);
+
     KRATOS_ERROR_IF(space_id < 0) << "H5Screate failed." << std::endl;
-    attr_id = H5Acreate_by_name(m_file_id, rObjectPath.c_str(), rName.c_str(), type_id,
-                                space_id, H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT);
+
+    type_id = Internals::GetPrimitiveH5Type<TDataType>();
+    attr_id = H5Acreate_by_name(m_file_id, rObjectPath.c_str(), rName.c_str(), type_id, space_id, H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT);
     KRATOS_ERROR_IF(attr_id < 0) << "H5Acreate_by_name failed." << std::endl;
-    KRATOS_ERROR_IF(H5Awrite(attr_id, type_id, rValue.c_str()) < 0) << "H5Awrite failed." << std::endl;
+    KRATOS_ERROR_IF(H5Awrite(attr_id, type_id, type_traits::GetContiguousData(rValue)) < 0) << "H5Awrite failed." << std::endl;
     KRATOS_ERROR_IF(H5Sclose(space_id) < 0) << "H5Sclose failed." << std::endl;
     KRATOS_ERROR_IF(H5Aclose(attr_id) < 0) << "H5Aclose failed." << std::endl;
+
     KRATOS_INFO_IF("HDF5Application", GetEchoLevel() == 2)
         << "Write time \"" << rObjectPath << '/' << rName
         << "\": " << timer.ElapsedSeconds() << std::endl;
-    KRATOS_CATCH("Path: \"" + rObjectPath + '/' + rName + "\".");
-}
 
-void File::WriteAttribute(const std::string& rObjectPath, const std::string& rName, const array_1d<double, 3>& rValue)
-{
-    KRATOS_TRY;
-    Vector<double> vector_value = rValue;
-    WriteAttribute(rObjectPath, rName, vector_value);
     KRATOS_CATCH("Path: \"" + rObjectPath + '/' + rName + "\".");
 }
 
-void File::WriteDataSet(const std::string& rPath, const Vector<int>& rData, WriteInfo& rInfo)
+void File::WriteAttribute(
+    const std::string& rObjectPath,
+    const Parameters Attributes)
 {
-    KRATOS_TRY;
-    KRATOS_ERROR << "Calling the base class method. Please override in the derived class." << std::endl;
-    KRATOS_CATCH("");
-}
+    KRATOS_TRY
 
-void File::WriteDataSet(const std::string& rPath, const Vector<double>& rData, WriteInfo& rInfo)
-{
-    KRATOS_TRY;
-    KRATOS_ERROR << "Calling the base class method. Please override in the derived class." << std::endl;
-    KRATOS_CATCH("");
-}
-
-void File::WriteDataSet(const std::string& rPath, const Vector<array_1d<double, 3>>& rData, WriteInfo& rInfo)
-{
-    KRATOS_TRY;
-    KRATOS_ERROR << "Calling the base class method. Please override in the derived class." << std::endl;
-    KRATOS_CATCH("");
-}
+    for (auto itr = Attributes.begin(); itr != Attributes.end(); ++itr) {
```

Comment:
y no range based `for`?

```suggestion
    for (auto attribute : Attributes) {
```

Reply:
this is not possible, since the name of the attribute is only available in the itr.